pkg/ddevapp/app_compose_template.yaml

```diff
@@ -162,11 +162,17 @@ services:
     hostname: {{ .Name }}-web
 
     ports:
-      - "{{ .DockerIP }}:$DDEV_HOST_HTTP_PORT:80"
-      - "{{ .DockerIP }}:$DDEV_HOST_HTTPS_PORT:443"
-    {{ if .HostMailpitPort }}
-      - "{{ .DockerIP }}:{{ .HostMailpitPort }}:8025"
-    {{ end }}
+    {{- if ne .WebserverType "generic" }}
+      - "{{ $.DockerIP }}:$DDEV_HOST_HTTP_PORT:80"
+      - "{{ $.DockerIP }}:$DDEV_HOST_HTTPS_PORT:443"
+    {{- end }}
+    {{- range .WebExtraContainerPorts }}
+      - "{{ $.DockerIP }}::{{ . }}"
```

Comment:
Nice, I didn't know that such a syntax was allowed with double `::`, where you skip the host port.